src/xopen/__init__.py

```diff
@@ -223,34 +224,37 @@ def __init__(  # noqa: C901
         if sys.platform != "win32":
             close_fds = True
 
+        self.in_pipe = None
+        self.in_thread = None
+        self._feeding = True
         if "r" in mode:
-            self._program_args += ["-c", "-d", path]  # type: ignore
-            self.outfile = None
+            self._program_args += ["-c", "-d"]  # type: ignore
+            stdout = subprocess.PIPE
+        else:
+            if compresslevel is not None:
+                self._program_args += ["-" + str(compresslevel)]
+            stdout = self.fileobj  # type: ignore
+        try:
             self.process = subprocess.Popen(
                 self._program_args,
                 stderr=self._stderr,
-                stdout=PIPE,
+                stdout=stdout,
+                stdin=subprocess.PIPE,
                 close_fds=close_fds,
             )  # type: ignore
+        except OSError:
+            if self.closefd:
+                self.fileobj.close()
+            raise
+        assert self.process.stdin is not None
+        if "r" in mode:
+            self.in_pipe = self.process.stdin
+            self.in_thread = threading.Thread(target=self._feed_pipe)
```

Comment:
Can you add a comment here or perhaps better to the `_feed_pipe` function why a thread is necessary/beneficial?